### PR TITLE
WIP: Squids More Unique

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
This is a fork from my previous change to glass shards, because squids slipped 100% of the time on them.

This adds a host of features to squids, including food tastes, sounds, reactions to salt (https://youtu.be/dxQmOR_QLfQ), more relevant buffs/nerfs to having tentacles and suction cups: Stronger grip, slower run speed, no slip trait, and less max melee damage.

This creates an innate ability for squids to change to a random color every 5 seconds, complete with custom, poorly drawn icon.

## Why It's Good For The Game
Squid people are normally just an appearance change. This makes them a bit more flavorful, thematic. 

## Changelog
:cl:
add: Squids spasm and take burn damage when touching salt
add: Squids scream and take burn damage when eating salt
add: Squids do not slip on surfaces, except glass shards (briefly)
add: Squiddly melee, grab, and death sounds
add: Ability to change colors
add: Ability icon that I shittily drew
add: Turn grey on death
tweak: Fried foods now toxic to squids
tweak: Squid's grip is slightly harder to resist out of
tweak: Squid melee max damage reduced 25%
tweak: Squids take much more elemental damage
tweak: Squids run slower
/:cl:

### CODE REVIEW

File: code/game/objects/effects/decals/cleanable/food.dm
Comment (ike709) [inline code/game/objects/effects/decals/cleanable/food.dm]: You're aware this will crit squids in a matter of seconds, right?

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (ike709) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: You don't actually check if it was ingested. Add `if(method == INGEST)`. I recommend adding a slightly different effect for `else if(method == TOUCH)`.

**Review by ike709 — CHANGES_REQUESTED** [review]
I'm okay with the idea of this PR , but there's still code/balancing concerns to be addressed.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (ike709) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: `H` should be `M`.

Comment (ike709) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: For all of these in this proc, FYI.

**Comment by jlsnow301** [comment]
Will readress soon

File: code/game/objects/effects/decals/cleanable/food.dm
Comment (ike709) [inline code/game/objects/effects/decals/cleanable/food.dm]: All of these `H`s need to be `L`s. `H` doesn't magically equal a human. Generally, you need to either use the var you have available or create a var for the human typepath and set it to the var you have available (then check it).

File: code/game/objects/effects/decals/cleanable/food.dm
Comment (jlsnow301) [reply code/game/objects/effects/decals/cleanable/food.dm]: Dropped slightly.

**Comment by qwertyquerty** [comment]
This seems to make squids substantially overpowered. Address that.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (ike709) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: Christ what the hell happened here? I don't even know where to begin with all of the crap you added to this proc.

File: code/game/objects/effects/decals/cleanable/food.dm
Comment (ike709) [inline code/game/objects/effects/decals/cleanable/food.dm]: Remove the second check for squid species and just make it a more generic message, something like "The salt! It burns!". Also this should be a `to_chat()`. 

You could also maybe add a `prob()` check to scream.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (ike709) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: Literally what?

Comment (jlsnow301) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: https://youtu.be/dxQmOR_QLfQ

Comment (ike709) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: I was referring to the fact that you duped the proc instead of just doing an `if(method == INGEST)` check.

Comment (jlsnow301) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: Thought that one of the parameters being method == touch meant this entire block was method==touch, and I would have to create a separate block for ingest. If I could just put method==ingest in it, easy

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (ike709) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: `=` sets a var equal to something, you want `==` which checks equality.

**Comment by ike709** [comment]
The more work you put into this, the more shitcode it becomes.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (ike709) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: This isn't a loop, this sleep does nothing.

Comment (jlsnow301) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: Interesting. Thought this was to apply sleep(time)

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (jlsnow301) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: Done

File: code/game/objects/effects/decals/cleanable/food.dm
Comment (jlsnow301) [reply code/game/objects/effects/decals/cleanable/food.dm]: This prob() check to scream is nice, will find it in code shortly

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (jlsnow301) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: Fuck me. Beginner (pro gamer) move

File: code/game/objects/effects/decals/cleanable/food.dm
Comment (jlsnow301) [reply code/game/objects/effects/decals/cleanable/food.dm]: Noted. Didn't see that define earlier.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (ike709) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: You can simplify all of this with:
`M.throw_item(get_ranged_target_turf(M, pick(GLOB.alldirs), rand(1, 3)))`
Then you can toss out the code involving picking a target.
(You'll still want to test this).

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (ike709) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: Yes but in this instance sleeping does nothing.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (jlsnow301) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: Might satisfy Travis also

**Comment by qwertyquerty** [comment]
are you even testing this code as you write it? because it shouldn't even compile

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (jlsnow301) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: Should I leave get_active_held_item() in here and do a check for it with if(I)?
Don't want to get a warning with no items

Comment (ike709) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: `throw_item()` already has the same sanity check in it.

**Comment by ike709** [comment]
@qwertyquerty He obviously isn't, and it doesn't, but step by step I'm transforming this from terrible code into decent code.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (jlsnow301) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: > are you even testing this code as you write it? because it shouldn't even compile
Ye, sorry, knee deep in just trying to implement things. Getting an inconsistent indentation on squidpeople.dm

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (ike709) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: `M` not `user`. You gotta stop pulling vars out of thin air.

**Comment by jlsnow301** [comment]
> @qwertyquerty He obviously isn't, but step by step I'm transforming this from terrible code into decent code.

Even a broken clock is right twice a day

**Comment by qwertyquerty** [comment]
Does broken code compile twice a day?

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (jlsnow301) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: fak

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (ike709) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: What's with the sleeping?

Comment (jlsnow301) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: Another thought was hallucinate. But I felt damage would be thematic for something that makes them spasm violently

Comment (ike709) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: Neither sleeping nor hallucinating makes sense here.

**Comment by ike709** [comment]
Hey, it probably compiles now.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (jlsnow301) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: Oh does it. Well then.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (jlsnow301) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: Black powder supposedly makes plasmemes hallucinate. Are you against FUN

Comment (ike709) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: In this case, yes.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (ike709) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm:333]: `I` no longer exists as a var.

**Comment by qwertyquerty** [comment]
Eggs moment

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (jlsnow301) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm:333]: Fixing now

**Review by ike709 — APPROVED** [review]
The code is probably as good as it's getting (at least, as good as it's getting without more time and effort than I'm willing to invest in it)

I leave balancing this in the hands of @qwertyquerty

File: code/modules/mob/living/living.dm
Comment (jlsnow301) [inline code/modules/mob/living/living.dm:755]: I could probably merge this into a long IF statement, but prefer the readability

Comment (ike709) [inline code/modules/mob/living/living.dm:755]: 🤔

Comment (jlsnow301) [reply code/modules/mob/living/living.dm:755]: Removed strong grabber for balance. This is a more tolerable version, just harder to break free.

**Comment by qwertyquerty** [comment]
It still seems like you're just trying to make squids better humans. They have two massive buffs and almost no nerfs

**Comment by zeskorion** [comment]
especially with the gay glass shard nerf.

**Comment by jlsnow301** [comment]
I will address both of your concerns soon. Though, the strong grabber trait is gone, I wouldn't say it's MASSIVE any more if it's just harder to break free from (doesn't actually give an agg grab on first click).

File: code/modules/mob/living/carbon/human/species_types/squidpeople.dm
Comment (ike709) [inline code/modules/mob/living/carbon/human/species_types/squidpeople.dm:45]: You need to add the following to `after_job_equip()` for your action to work:
```
var/datum/action/innate/squid_change/S = new
S.Grant(H)
```

File: code/modules/mob/living/carbon/human/species_types/squidpeople.dm
Comment (ike709) [inline code/modules/mob/living/carbon/human/species_types/squidpeople.dm]: This action var does nothing. Remove it. See my other comment.

File: code/modules/mob/living/carbon/human/species_types/squidpeople.dm
Comment (ike709) [inline code/modules/mob/living/carbon/human/species_types/squidpeople.dm:70]: I'm not sure if actions have built-in spam prevention, so test if it does or not and I'll show you how to add it if it doesn't.

Comment (ike709) [reply code/modules/mob/living/carbon/human/species_types/squidpeople.dm:70]: Either way it may be worth putting a 5 second delay between uses, tbh.

Comment (jlsnow301) [reply code/modules/mob/living/carbon/human/species_types/squidpeople.dm:70]: I agree, I tried briefly (before getting burned out) finding the code but couldn't find any type of cooldown for innate abilities like this

**Comment by qwertyquerty** [comment]
Check travis logs and fix linting issues

**Comment by jlsnow301** [comment]
Still having issues with the cooldown on an innate ability. Should pass tests now.

File: code/modules/mob/living/carbon/human/species_types/squidpeople.dm
Comment (ike709) [inline code/modules/mob/living/carbon/human/species_types/squidpeople.dm:49]: Put this in here too:
```
var/datum/action/innate/squid_change/S = locate(/datum/action/innate/squid_change) in M.actions
S?.Remove(M)
```

File: code/modules/mob/living/carbon/human/species_types/squidpeople.dm
Comment (ike709) [inline code/modules/mob/living/carbon/human/species_types/squidpeople.dm]: I wonder if there's a GLOB somewhere with a list of colors.

**Comment by ike709** [comment]
This PR is bloated enough. No new features, only code improvements and balancing. Save everything else for a future PR.

File: code/modules/mob/living/carbon/human/species_types/squidpeople.dm
Comment (ike709) [inline code/modules/mob/living/carbon/human/species_types/squidpeople.dm]:
```suggestion
    return ..()
```

Pretty sure you need this so it runs the parent's sanity checks (assuming there are any, too tired to look).

File: code/modules/mob/living/carbon/human/species_types/squidpeople.dm
Comment (ike709) [reply code/modules/mob/living/carbon/human/species_types/squidpeople.dm]: Having a huge switch statement for semi-random colors is kind of gross.

Comment (jlsnow301) [reply code/modules/mob/living/carbon/human/species_types/squidpeople.dm]: Yeah it's getting pretty ridic. This code block was pulled from xenobio/crossbreeding/__corecross.dm. It's still only giving me white, however.

**Comment by jlsnow301** [comment]
Agreed. Once color change is done, I'm finished with it.

File: code/modules/mob/living/carbon/human/species_types/squidpeople.dm
Comment (jlsnow301) [reply code/modules/mob/living/carbon/human/species_types/squidpeople.dm:49]: Done

Comment (jlsnow301) [reply code/modules/mob/living/carbon/human/species_types/squidpeople.dm:49]: What is M here? Getting an undefined var

**Comment by qwertyquerty** [comment]
If your changelog is up to date, this seems fairly balanced to me now.

**Comment by jlsnow301** [comment]
It is, just having trouble with the change color ability working: It's just giving me white. Not sure if the rand(1,20) is not working or if it's just reverting to #FFFFFF.

**Comment by qwertyquerty** [comment]
why did you remove metacoin again

**Comment by jlsnow301** [comment]
Snuck into commits earlier. Didn't mean to include it. Didn't know how to just discard the file change, tbh

**Comment by qwertyquerty** [comment]
Put it back

File: code/modules/mob/living/carbon/human/species_types/squidpeople.dm
Comment (jlsnow301) [reply code/modules/mob/living/carbon/human/species_types/squidpeople.dm:49]: I can't find this, reverting for now

File: code/modules/mob/living/carbon/human/species_types/squidpeople.dm
Comment (jlsnow301) [reply code/modules/mob/living/carbon/human/species_types/squidpeople.dm]: Fixed

**Comment by jlsnow301** [comment]
Need last touches on it:
a) Icon update not working, have to click or hide/show it to update icon
b) Diffs is not happy

**Comment by ike709** [comment]
Diff bot is broken, ignore it.

File: code/modules/mob/living/carbon/human/species_types/squidpeople.dm
Comment (Wilchenx) [inline code/modules/mob/living/carbon/human/species_types/squidpeople.dm:22]: this is a spongebob refrence, there is really no reason to change it to toxic

**Comment by qwertyquerty** [comment]
Ignore icondiffbot. Is this ready?

**Comment by jlsnow301** [comment]
It's ready. Needs a look eventually to resolve the icon update issue. But everything else okay.

File: code/modules/mob/living/carbon/human/species_types/squidpeople.dm
Comment (qwertyquerty) [inline code/modules/mob/living/carbon/human/species_types/squidpeople.dm]: These could probably be in a static list for the species type.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (qwertyquerty) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: Please pad == with spaces

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (qwertyquerty) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: Ditto

**Comment by qwertyquerty** [comment]
I'll wait for the @ike709 seal of approval before putting this to vote, but this all looks good to me balance and codewise.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm
Comment (jlsnow301) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: done

Comment (qwertyquerty) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: Whatever you did you didn't push

File: code/modules/mob/living/carbon/human/species_types/squidpeople.dm
Comment (jlsnow301) [reply code/modules/mob/living/carbon/human/species_types/squidpeople.dm]: Throws an undefined var if I move the elsewhere, why even use one, I thought

**Comment by qwertyquerty** [comment]
I'm putting this up to vote but I still await ikes seal for merging

File: code/modules/mob/living/carbon/human/species_types/squidpeople.dm
Comment (ike709) [reply code/modules/mob/living/carbon/human/species_types/squidpeople.dm:49]: My bad. `H` not `M`. You should be able to figure this out by now though. And this still needs to be done.

**Review by ike709 — CHANGES_REQUESTED** [review]
Do the thing I said to do for the action to get removed on species loss, then this is ready.

**Comment by qwertyquerty** [comment]
So is this ready @ike709 ?

**Comment by ike709** [comment]
Assuming it's been tested, yes.

**Comment by jlsnow301** [comment]
Hosted privately. Feels nice, icon just a little wonky but manageable (all icons seem this way tbf)

**Review by ike709 — APPROVED** [review]

**Pull request merged**
